webui/react/src/hooks/useSettings.ts

```diff
@@ -315,7 +316,7 @@ const useSettings = <T>(config: SettingsConfig, options?: SettingsHookOptions):
   }, [ config.settings, updateSettings ]);
 
   const fetchUserSetting = useCallback(async () => {
-    if (!user) return;
+    if (!user || user === prevUser) return;
```

Comment:
`getUserSetting` is called based on `user.id`, so check if `user` is the same or not. If its the same user, return early in function.